packages/fuel-indexer-tests/src/lib.rs

```diff
@@ -29,6 +29,12 @@ pub mod defaults {
     pub const WALLET_PASSWORD: &str = "password";
     pub const INDEXED_EVENT_WAIT: u64 = 2;
     pub const COIN_AMOUNT: u64 = 11;
+
+    // Note: We need the contract ID in order to connect an instance to
+    // the node provider for E2E tests, but spawning a task that is supposed
+    // to return the contract ID won't give us the ID until the task is completed.
+    pub const CURRENT_TEST_CONTRACT_ID_STR: &str =
```

Comment:
@deekerno 
- Is there any reason we're hardcoding this in a second place vs just connecting to the contract to get the contract ID? 
- I'm assuming there was some change in the dependencies now that prevents us from connecting to the contract like we were?

Reply:
Yes, you're correct in your assumption. Previously, we were using `get_contract_id` which was actually redeploying the contract under-the-hood. However, the underlying `fuel-core` dependency does not allow that anymore; instead, it returns a `ContractAlreadyDeployed` error. I attempted to work around this by spawning the fixture and then returning the ID for the deployed contract and the join handle for the node fixture; however, the only way that I would be able to get the ID from `Contract::deploy(..)` from a JoinHandle is if the task completed fully, which means that it wouldn't be running for the tests.